src/main/kotlin/net/emdal/tank/Query.kt

```diff
@@ -1,25 +1,17 @@
 package net.emdal.tank
 
-import net.emdal.tank.Clause.CREATE
-import net.emdal.tank.Clause.MATCH
-import org.neo4j.driver.Session
+import net.emdal.tank.clause.Clause
+import net.emdal.tank.clause.CreateClause
+import net.emdal.tank.clause.MatchClause
 
-fun Session.query(block: Graph.() -> Graph) =
-  this.run("${Graph().block().query} RETURN *").list().toList()
+data class Query(val clauses: List<Clause> = emptyList()) {
+  val query get() = clauses.joinToString("\n") { it.query.joinToString("") }
```

Comment:
Nit: Naming things like Data class `Query` with a field `query`? Since this getter actually fetches the cypher string perhaps it should be renamed to `cypher`?